core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/IoUtil.java

```diff
@@ -136,6 +143,36 @@ public static InputStream getResource(String resource) throws IOException {
         return is;
     }
 
+    /**
+     * Lists all children resources.
+     *
+     * @param parent
+     *      the root path represented in {@link URI} format
+     * @throws java.io.IOException
+     *      if any i/o error occur
+     * @throws ProviderNotFoundException
+     *      if a provider supporting the URI scheme is not installed
+     */
+    public static void listResources(URI parent, Consumer<Path> consumer) throws IOException {
+        FileSystem fileSystem = null;
+        try {
+            if (!parent.getScheme().equals("file")) {
```

Comment:
If it is possible that provided URI is relative and this line throws NPE, argument can be used on the left side instead